doc/source/fault-tolerance.rst

```diff
@@ -88,3 +88,87 @@ You can experiment with this behavior by running the following code.
             print(counter)
         except ray.exceptions.RayActorError:
             print('FAILURE')
+
+Actor Checkpointing
+~~~~~~~~~~~~~~~~~~~
+
+It's often important to be able to checkpoint the state of an actor in order to
+recover that state in the event of a failure. This can be done by having the
+actor inherit from the ``ray.actor.Checkpointable`` interface.
+
+Several things are left to the application and are not done by Ray. The reason
+for this is that applications will want to handle checkpointing in many
+different ways.
+- When to create checkpoints. The ``should_checkpoint`` method should determine
+  this.
+- What to checkpoint, how to serialize it, and where to store the checkpoint.
+  The ``save_checkpoint`` method handles this.
+- How to restore the actor state from a given checkpoint. The
+  ``load_checkpoint`` method takes care of this.
```

Comment:
For the user reading documentation, I think this should be written in an actionable tone. Reasoning and design decisions are good though they should be not be the primary focus. For example:

- Override ``save_checkpoint`` to specify what to checkpoint, how to serialize it, and where to store the checkpoint.
*provide example*

- Override ``load_checkpoint`` to specify how to restore from a given checkpoint.
- Override ``should_checkpoint`` to programmatically specify the frequency of which ``save_checkpoint`` is called.